pkg/proto/common.go

```diff
@@ -76,11 +77,15 @@ func StringWithUInt32Len(buf []byte) (string, error) {
 	return r, nil
 }
 
-// PutBytesWithUInt16Len prepends given buf with 2 bytes of it's length.
-func PutBytesWithUInt16Len(buf []byte, data []byte) {
-	sl := uint16(len(data))
-	binary.BigEndian.PutUint16(buf, sl)
+// PutBytesWithUInt16Len prepends given buf with 2 bytes of its length.
+func PutBytesWithUInt16Len(buf []byte, data []byte) error {
+	l := len(data)
+	if l > math.MaxInt16 {
+		return errors.Errorf("invalid data size %d", l)
+	}
+	binary.BigEndian.PutUint16(buf, uint16(l))
 	copy(buf[2:], data)
```

Comment:
I think `len(buf) >= 2 + len(data)` is necessary here.

Reply:
Yes, you are right, I suppose those checks was left out because the earlier version of this function doesn't return an error.

Reply:
added